storage/host-flash/host-flash.c

```diff
@@ -15,155 +15,167 @@
+
+		readed = hostflash_read(addr + wrote, tempTab, len);
+		if (readed <= 0)
+			return readed;
+
+		for (i = 0; i < readed; i++)
+			tempTab[i] = tempTab[i] & *((char *)buff + wrote + i);
+
+		len = 0;
+		while (len != readed) {
+			stat = pwrite(hostflash_common.filefd, &tempTab, readed - len, addr + wrote + len);
+			if (stat < 0) {
+				if (errno == EINTR)
+					continue;
+
+				return stat;
+			}
+			len += stat;
+		}
+		wrote += len;
+	}
+
+	return (ssize_t)wrote;
 }
 
 
 void hostflash_sectorErase(unsigned int addr)
 {
-    char tempTab[256];
-    ssize_t len = sizeof(tempTab);
-    int erased = 0;
-    unsigned int sectorNum;
-    unsigned int sectorAddr;
-    int stat;
-
-    memset(tempTab, 0xFF, sizeof(tempTab));
-
-    if (addr < hostflash_common.flashsz) {
-        sectorNum = addr / hostflash_common.sectorsz;
-        sectorAddr = sectorNum * hostflash_common.sectorsz;
-    }
-    else {
-        return;
-    }
-
-    while (erased != hostflash_common.sectorsz) {
-        stat = pwrite(hostflash_common.filefd, tempTab, len, sectorAddr + erased);
-        if (stat < 0) {
-            if (errno == -EINTR)
-                continue;
-
-            return;
-        }
-        else {
-            len = len - stat;
-            if (len == 0)
-                len = sizeof(tempTab);
-        }
-
-        erased += stat;
-    }
-
-    return;
+	char tempTab[256];
+	ssize_t len = sizeof(tempTab);
+	int erased = 0;
+	unsigned int sectorNum;
+	unsigned int sectorAddr;
+	int stat;
+
+	memset(tempTab, 0xFF, sizeof(tempTab));
+
+	if (addr < hostflash_common.flashsz) {
+		sectorNum = addr / hostflash_common.sectorsz;
+		sectorAddr = sectorNum * hostflash_common.sectorsz;
+	}
+	else {
+		return;
+	}
+
+	while (erased != hostflash_common.sectorsz) {
+		stat = pwrite(hostflash_common.filefd, tempTab, len, sectorAddr + erased);
+		if (stat < 0) {
+			if (errno == EINTR)
+				continue;
+
+			return;
+		}
+		else {
+			len = len - stat;
+			if (len == 0)
+				len = sizeof(tempTab);
+		}
+
+		erased += stat;
+	}
+
+	return;
 }
 
 
 void hostflash_chipErase(void)
 {
-    char tempTab[256];
-    size_t erased = 0, len = sizeof(tempTab);
-    ssize_t stat;
-
-    memset(tempTab, 0xFF, sizeof(tempTab));
-
-    while (erased != hostflash_common.flashsz) {
-        stat = pwrite(hostflash_common.filefd, tempTab, len, erased);
-        if (stat < 0) {
-            if (errno == -EINTR)
-                continue;
-
-            return;
-        }
-        else {
-            len = len - stat;
-            if (len == 0)
-                len = sizeof(tempTab);
-        }
-
-        erased += stat;
-    }
-
-    return;
+	char tempTab[256];
+	size_t erased = 0, len = sizeof(tempTab);
+	ssize_t stat;
+
+	memset(tempTab, 0xFF, sizeof(tempTab));
+
+	while (erased != hostflash_common.flashsz) {
+		stat = pwrite(hostflash_common.filefd, tempTab, len, erased);
+		if (stat < 0) {
+			if (errno == EINTR)
+				continue;
+
+			return;
+		}
+		else {
```

Comment:
In the `if` before we either `continue` or `return`, no need for `else`

Reply:
Done